components.py

```diff
@@ -307,6 +307,10 @@ def add_component(obj, component_name, hubs_config, registered_hubs_components):
             else:
                 component[property_name] = default_value
 
+    if 'deps' in component_definition:
+        for dep in component_definition["deps"]:
+            add_component(obj, dep, hubs_config, registered_hubs_components)
```

Comment:
This doesn't check if the object already has this component.. It might make sense to just do that in add_component itself. Or maybe both (with add_component logging a warning)

Reply:
I've added a check to avoid duplicated components